### PR TITLE
Initialize unit and integration tests and create github CI

### PR DESCRIPTION
**Pull request by tomitheninja**
- Fix imports are not working in `src/**/*.spec.ts` files
- Fix eslint prettier integration
- Configure eslint to run on test files as well
- Force mock prisma and passport modules to avoid side effects
- Write tests to verify database and passport modules are mocked
- Add integration test for JWT token creation
- Add integration test for active vulnerability: _users can change anyone's authSchId_ in [userController.update](https://github.com/kir-dev/konzisite-api/blob/37f58498803b8e0423eae5b1f998ddc94272cb37/src/users/users.controller.ts#L49)
- Create github workflow to check code quality on pull requests

### CODE REVIEW

**Review by Tschonti — APPROVED** [review]
Thanks for this, the vulnerability will be fixed soon!

**Review by triszt4n — COMMENTED** [review]
Wow, cool so far, much appreciated! Gonna test it on my machine, then we can merge!

File: .eslintrc.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 module.exports = {
   parser: '@typescript-eslint/parser',
   parserOptions: {
-    project: 'tsconfig.json',
+    project: 'linter.tsconfig.json',
```
Comment (triszt4n) [inline .eslintrc.js:4]: This file could be rather called `tsconfig.linter.json` like `tsconfig.build.json` in the root folder. But this is not really an issue, this is not important

**Review by triszt4n — APPROVED** [review]
All good, let's merge @Tschonti

**Pull request merged**